crates/sqllogictest/src/schedule.rs

```diff
@@ -18,14 +18,26 @@
 use std::collections::HashMap;
 use std::fs::read_to_string;
 use std::path::{Path, PathBuf};
+use std::sync::Arc;
 
 use anyhow::{Context, anyhow};
+use iceberg::Catalog;
 use serde::{Deserialize, Serialize};
 use toml::{Table as TomlTable, Value};
 use tracing::info;
 
 use crate::engine::{EngineRunner, load_engine_runner};
 
+/// Configuration for a catalog in the schedule file.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct CatalogConfig {
```

Comment:
Catalog config should be engine specific, e.g., it should be part of an engine, rather in a common section.